File: javaproject/app/src/main/java/com/marcosvbras/robomarket/flows/createsale/viewmodel/CreateSaleViewModelFactory.java
package com.marcosvbras.robomarket.flows.createsale.viewmodel;

import android.arch.lifecycle.ViewModel;
import android.arch.lifecycle.ViewModelProvider;
import android.support.annotation.NonNull;

import com.marcosvbras.robomarket.interfaces.BaseActivityCallback;

import javax.inject.Inject;

public class CreateSaleViewModelFactory implements ViewModelProvider.Factory {

    private final BaseActivityCallback activityCallback;

    @Inject
    public CreateSaleViewModelFactory(BaseActivityCallback activityCallback) {
        this.activityCallback = activityCallback;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new CreateSaleViewModel(activityCallback);
    }
}

File: javaproject/app/src/main/java/com/marcosvbras/robomarket/flows/createsale/viewmodel/CreateSaleViewModelFactory.kt
package com.marcosvbras.robomarket.flows.createsale.viewmodel

import android.arch.lifecycle.ViewModel
import android.arch.lifecycle.ViewModelProvider
import com.marcosvbras.robomarket.interfaces.ActivityCallback

@Suppress("UNCHECKED_CAST")
class CreateSaleViewModelFactory(private val activityCallback: ActivityCallback) : ViewModelProvider.NewInstanceFactory() {

    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        return CreateSaleViewModel(activityCallback) as T
    }
}